src/mission_ui.cpp

```diff
@@ -17,239 +17,277 @@
-        } else if( action == "CONFIRM" ) {
-            if( tab == tab_mode::TAB_ACTIVE && selection < umissions.size() ) {
-                u.set_active_mission( *umissions[selection] );
-            }
-            break;
-        } else if( action == "SELECT" ) {
-            // get clicked coord
-            std::optional<point> coord = ctxt.get_coordinates_text( w_missions );
-            if( coord.has_value() ) {
-
-                for( auto &it : tabs_coords ) {
-                    if( it.second.contains( coord.value() ) ) {
-                        tab = it.first;
-                    }
-                }
-
-                for( auto &it : mission_row_coords ) {
-                    if( it.second.contains( coord.value() ) ) {
-                        selection = it.first;
-                        if( tab == tab_mode::TAB_ACTIVE && selection < umissions.size() ) {
-                            u.set_active_mission( *umissions[selection] );
-                        }
-                    }
-                }
-            }
-        } else if( action == "MOUSE_MOVE" ) {
-            // get clicked coord
-            std::optional<point> coord = ctxt.get_coordinates_text( w_missions );
-            if( coord.has_value() ) {
-
-                for( auto &it : mission_row_coords ) {
-                    if( it.second.contains( coord.value() ) ) {
-                        selection = it.first;
-                    }
-                }
+        parse_tags( parsed_description, get_player_character(), get_player_character() );
+    }
+    draw_colored_text( parsed_description, c_unset, table_column_width * 1.15 );
+    if( miss->has_deadline() ) {
+        const time_point deadline = miss->get_deadline();
+        ImGui::Text( _( "Deadline: %s" ), to_string( deadline ).c_str() );
+        if( selected_tab == mission_ui_tab_enum::ACTIVE ) {
+            // There's no point in displaying this for a completed/failed mission.
+            // @ TODO: But displaying when you completed it would be useful.
+            const time_duration remaining = deadline - calendar::turn;
+            std::string remaining_time;
+            if( remaining <= 0_turns ) {
+                remaining_time = _( "None!" );
+            } else if( get_player_character().has_watch() ) {
+                remaining_time = to_string( remaining );
+            } else {
+                remaining_time = to_string_approx( remaining );
             }
-        } else if( action == "QUIT" ) {
-            break;
+            ImGui::TextWrapped( _( "Time remaining: %s" ), remaining_time.c_str() );
         }
     }
+    if( miss->has_target() ) {
+        // TODO: target does not contain a z-component, targets are assumed to be on z=0
+        const tripoint_abs_omt pos = get_player_character().global_omt_location();
+        draw_colored_text( string_format( _( "Target: %s" ), miss->get_target().to_string() ), c_white );
```

Comment:
Since these will never contain <color> tags, you could just use `ImGui::Text`, which is much simpler.

Reply:
The intention here was to specifically force them to be white (or register as c_white, anyway), mimicking the old coloration. It is possible that the imgui default text color will be changing or otherwise adjustable settings options will be added, so this was just a bit of forward-thinking.